### PR TITLE
push message

### PR DESCRIPTION
**Pull request by tri-et**

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -22,6 +22,7 @@ module.exports = {
     __statics: true,
     importScripts: true,
     workbox: true,
+    clients: true,
```
Comment (hemvanh) [inline .eslintrc.js:25]: purpose of this ?

File: quasar.conf.js [hunk 2]
```diff
@@ -120,6 +120,7 @@ module.exports = function(ctx) {
       // workboxPluginMode: 'GenerateSW',
       // workboxOptions: {},
       manifest: {
+        gcm_sender_id: '889927801023',
```
Comment (hemvanh) [inline quasar.conf.js:123]: where to get this Id, is it hard-coded and work forever ?

File: server/schema/customer/res.js [hunk 3]
```diff
@@ -40,6 +40,11 @@ const resolvers = {
       if (user) return jwt.sign(`${input}_${new Date().getTime()}`, process.env.JWT_SECRET)
       else throw new Error('Customer not found!')
     },
+    async registerPushSubscription(_, {input}) {
+      return await PushSubscription.create(input).then(async () => {
```
Comment (hemvanh) [inline server/schema/customer/res.js]: async ()  => {}  --> why need async ????

File: server/schema/admin/res.js [hunk 4]
```diff
@@ -43,6 +63,29 @@ const resolvers = {
         })
       }
     },
+    async pushMessage(_, {input}, {loggedInUser}) {
+      _authAdmin(loggedInUser)
+      let data = await PushSubscription.findAll()
+      let pushComlepted = Promise.resolve()
```
Comment (hemvanh) [inline server/schema/admin/res.js]: pushComlepted ---> what is it for ???
pls find a proper way of using synced promise ....
try: https://caolan.github.io/async/
or proper native Promise implementation

File: src-pwa/custom-service-worker.js [hunk 5]
```diff
@@ -22,6 +22,20 @@ self.addEventListener('fetch', event => {
...
-
+self.addEventListener('push', function(e) {
+  var options = {
+    body: e.data.text(),
+    icon: 'statics/icons/apple-icon-152x152.png',
+  }
+  e.waitUntil(self.registration.showNotification('TPTEA', options))
+})
+self.addEventListener('notificationclick', function(event) {
+  self.registration.getNotifications().then(function(notifications) {
```
Comment (hemvanh) [inline src-pwa/custom-service-worker.js]: why need to close all notifications ??

File: src-pwa/register-service-worker.js [hunk 6]
```diff
@@ -4,13 +4,47 @@
...
+}
 register(process.env.SERVICE_WORKER_FILE, {
   ready() {
     console.log('App is being served from cache by a service worker.')
   },
   registered(registration) {
     // registration -> a ServiceWorkerRegistration instance
     console.log('Service worker has been registered.')
+    registration.pushManager.getSubscription().then(sub => {
+      if (sub === null) {
```
Comment (hemvanh) [inline src-pwa/register-service-worker.js]: if(sub) or if(!sub)

File: package-lock.json [hunk 7]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "tptea",
-  "version": "0.0.1",
+  "version": "0.0.2",
```
Comment (hemvanh) [inline package-lock.json:3]: pls exclude your packpage-lock.json

**Review by hemvanh — CHANGES_REQUESTED** [review]

File: quasar.conf.js (hunk 2 above)
Comment (tri-et) [reply quasar.conf.js:123]: this id we got from https://console.firebase.google.com,
register account and create project

File: .eslintrc.js (hunk 1 above)
Comment (tri-et) [reply .eslintrc.js:25]: for eslint  dont show error like workbox

File: quasar.conf.js (hunk 2 above)
Comment (hemvanh) [reply quasar.conf.js:123]: so this using Firebase push ?

File: src-pwa/custom-service-worker.js (hunk 5 above)
Comment (tri-et) [reply src-pwa/custom-service-worker.js]: when we click one of all notification it'll show all notification in list so i close all notifications

**Review by hemvanh — APPROVED** [review]

**Pull request merged**